use_redhat and migration link (#462)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/con_observability-strategy.adoc

= Observability Strategy in Service Telemetry Framework

[role="_abstract"]

> {Project} ({ProjectShort}) does not include event storage backends or dashboarding tools. {ProjectShort} can optionally use community operators to deploy Elasticsearch and Grafana for those purposes. {ProjectShort} makes requests to these community operators to create instances of each application configured to work with {ProjectShort}.

Reviewer: ```suggestion
{Project} ({ProjectShort}) does not include event storage backends or dashboarding tools. {ProjectShort} can optionally create datasource configurations for Grafana using the community operator  to provide a dashboarding interface.
```

Reviewer: @csibbitt how about that rewording?

Instead of having Service Telemetry Operator create custom resource requests, you can use your own deployments of these applications or other compatible applications, and scrape the metrics Smart Gateways for delivery to your own Prometheus-compatible system for telemetry storage. If you set the `observabilityStrategy` to `none`, then storage backends will not be deployed so persistent storage will not be required by {ProjectShort}.

Use the observabilityStrategy property on the {ProjectShort} object to specify which type of observability components will be deployed.

The following values are available:

| value | meaning |
|-------|---------|

> | use_redhat | Only Red Hat supported components are requested by {ProjectShort}. This includes Prometheus and Alertmanager from the Observability Operator, but no resource requests to Elasticsearch Operator or Grafana Operator. |

Reviewer: ```suggestion
| use_redhat | Red Hat supported components are requested by {ProjectShort}. This includes Prometheus and Alertmanager from the Observability Operator, but no resource requests to Elastic Cloud on Kubernetes (ECK) Operator. If enabled, resources are also requested from the Grafana Operator (community component). |
```

Reviewer: Just FYI I called this out to Elastic Cloud on Kubernetes vs Elasticsearch Operator, which also exists, but is different (used for OpenShift Logging, historically).

= Observability Strategy in Service Telemetry Framework

[role="_abstract"]

> {Project} ({ProjectShort}) does not include event storage backends or dashboarding tools. {ProjectShort} can optionally use community operators to deploy Elasticsearch and Grafana for those purposes. {ProjectShort} makes requests to these community operators to create instances of each application configured to work with {ProjectShort}.

Author: This still makes it sound like we will make requests to ECK. I'd just leave off the second sentence maybe?

Instead of having Service Telemetry Operator create custom resource requests, you can use your own deployments of these applications or other compatible applications, and scrape the metrics Smart Gateways for delivery to your own Prometheus-compatible system for telemetry storage. If you set the `observabilityStrategy` to `none`, then storage backends will not be deployed so persistent storage will not be required by {ProjectShort}.

> Use the observabilityStrategy property on the {ProjectShort} object to specify which type of obvservability components will be deployed.

Reviewer: ```suggestion
Use the observabilityStrategy property on the {ProjectShort} object to specify which type of observability components will be deployed.
```

The following values are available:

| value | meaning |
|-------|---------|
| use_redhat | Only Red Hat supported components are requested by {ProjectShort}. This includes Prometheus and Alertmanager from the Observability Operator, but no resource requests to Elasticsearch Operator or Grafana Operator. |

> | use_hybrid | In addition to the Red Hat supported components, Elasticsearch and Grafana resources are also requested (if specified in the ServiceTelemetry object) |

Reviewer: @csibbitt any suggestions for how we should reword this section?

Maybe something like...

> Only Red Hat supported components are requested by {ProjectShort}. This includes Prometheus and Alertmanager from the Observability Operator, but no resource requests to Elasticsearch Operator or Grafana Operator.

to

> Red Hat supported components are requested by {ProjectShort}. This includes Prometheus and Alertmanager from the Observability Operator, but no resource requests to Elasticsearch Operator. If enabled, resources are also requested from the Grafana Operator (community component).

Author: Sure, that will do.

> | use_community | The community version of Prometheus Operator is used instead of Observability Operator. Elasticsearch and Grafana resources are also requested (if specified in the STF object) |

Reviewer: ```suggestion
| use_community | The community version of Prometheus Operator is used instead of Observability Operator. Elasticsearch and Grafana resources are also requested (if specified in the ServiceTelemetry object) |
```

| none | No storage or alerting components are deployed |

Newly deployed STF environments as of 1.5.3 default to `use_redhat`.

> Existing STF deployments created before 1.5.3 default to `use_community`.

Reviewer: ```suggestion
Existing {ProjectShort} deployments created before 1.5.3 default to `use_community`.
```

> Newly deployed {ProjectShort} environments as of 1.5.3 default to `use_redhat`.
> 
> Existing {ProjectShort} deployments created before 1.5.3 default to `use_community`.
> 
> To migrate an existing {ProjectShort} deployment to `use_redhat`, see https://access.redhat.com/articles/7011708[Migrating STF to fully supported operators]

Reviewer: ```suggestion
Newly deployed {ProjectShort} environments as of 1.5.3 default to `use_redhat`. Existing {ProjectShort} deployments created before 1.5.3 default to `use_community`.

To migrate an existing {ProjectShort} deployment to `use_redhat`, see https://access.redhat.com/articles/7011708[Migrating STF to fully supported operators].
```

Reviewer: Just pulled this up into a single line because it all seems related, but I could be convinced it looks better in separate lines.
